src/components/App.js

```diff
@@ -1,29 +1,73 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import { useQuery } from '@apollo/client';
 import { GET_ALL_ITEMS } from '../api';
+import { useState, useEffect } from 'react';
+import Landing from './Landing';
+import Products from './Products';
+import About from './About';
 
 function App() {
-  const { loading, error, data } = useQuery(GET_ALL_ITEMS)
+  const [items, setItems] = useState([])
+  const [itemsForDisplay, setItemsForDisplay] = useState([])
 
-  let allItems
-
-  if (!loading) {
-    console.log('data:', data)
-    allItems = data.queryItems.map(item => {
-      return <div>
-        <h1>{item.name}</h1>
-        <p>{item.description}</p>
-        <p>{item.color}</p>
-        <p>{item.price}</p>
-      </div>
-    })
+  const getItemsForDisplay = () => {
+    let productsForDisplay = [];
+    
+    const productNames = new Set (items.map(product => product.name))
+    
+    productNames.forEach(productName => {
+      
+      const filteredProducts = items
+        .filter(product => product.name === productName)
+      
+      const filteredProduct = filteredProducts.reduce((acc, curr) => {
+        acc.id = curr.name;
+        acc.name = curr.name;
+        acc.price = curr.price;
+        acc.category = curr.category
+        acc.image = curr.image
+        acc.description = curr.description
+        if (!acc.colorOptions.includes(curr.color)) {
+          acc.colorOptions.push(curr.color)
+        }
+        if (!acc.sizeOptions.includes(curr.size)) {
+          acc.sizeOptions.push(curr.size)
+        }
+        return acc;
+      }, {
+        colorOptions: [],
+        sizeOptions: []
+      })
+      productsForDisplay.push(filteredProduct)
+    }) 
+    return productsForDisplay
```

Comment:
This is a solid cleanup function. Nicely done. In a future branch, if we have different images per color, we can implement that as an attribute in the acc too.